File: php/Program/Shirt.php
<?php
// Shirt.php

include 'Clothing.php'; // Mengimpor kelas Clothing untuk mewarisi atribut dan metode

class Shirt extends Clothing { // Mendefinisikan kelas Shirt yang merupakan turunan dari kelas Clothing
    protected $color; // Properti untuk menyimpan warna kemeja
    protected $sleeveType; // Properti untuk menyimpan tipe lengan kemeja

    public function setColor($color) { // Metode untuk mengatur warna kemeja
        $this->color = $color; // Mengatur nilai properti color dengan nilai yang diberikan
    }

    public function setSleeveType($sleeveType) { // Metode untuk mengatur tipe lengan kemeja
        $this->sleeveType = $sleeveType; // Mengatur nilai properti sleeveType dengan nilai yang diberikan
    }

    public function getColor() { // Metode untuk mendapatkan warna kemeja
        return $this->color; // Mengembalikan nilai properti color
    }

    public function getSleeveType() { // Metode untuk mendapatkan tipe lengan kemeja
        return $this->sleeveType; // Mengembalikan nilai properti sleeveType
    }
}
?>

File: python/Program/shirt.py
from clothing import Clothing  # Import kelas Clothing untuk mewarisi atribut dan metode

class Shirt(Clothing):  # Mendefinisikan kelas Shirt yang merupakan turunan dari Clothing
    def __init__(self, idProduct, name, brand, price, size, material, gender, color, sleeve_type):
        super().__init__(idProduct, name, brand, price, size, material, gender)  # Memanggil konstruktor kelas induk (Clothing)
        self.color = color  # Inisialisasi atribut color
        self.sleeve_type = sleeve_type  # Inisialisasi atribut sleeve_type

    def display_info(self):
        return super().display_info() + [self.color, self.sleeve_type]  # Mengembalikan informasi produk, termasuk informasi tambahan dari kelas Shirt
